src/site/topic6.rst

```diff
@@ -98,12 +98,44 @@ Implementation
     * The other will allow us ot provide a starting size of the array
 
 
+.. code-block:: Java
+    :linenos:
+    :emphasize-lines: 2, 9
+
+        public ArrayStack() {
+            this(DEFAULT_CAPACITY);
+        }
+
+        public ArrayStack(int size) {
+            top = 0;
+            // Generic types cannot be instantiated so we cast
+            // This does generate a compile time warning
+            stack = (T[]) new Object[size];
+        }
+
+* You will see what we are doing overloading and  :doc:`constructor chaining </topic3-chaining>`
+* You will also see that we are creating an array of type ``Object`` and then *casting* it to the generic type `T`
+    * We cannot actually instantiate (create) a generic type
+    * This includes instantiating arrays of the generic type like above
```

Comment:
Fun fact, you actually can make a generic type, but it's reflection magic. And, that's not the reason you can't make a generic array. It actually has to do with the array type hierarchy. Consider a blob like:
```java
T[] ts = new T[2];
Object[] os = ts;  // legal statement, arrays are covariant
os[0] = "sup";  // legal statement, assigning String to an Object
```
`os` is actually an `T[]` and you can't assign a `String` to a `T`, so it'd result in a `ClassCastException` at _runtime_ -- arrays are covariant, but generic types are invariant. Very evil stuff, so java forbids it entirely.

Reply:
I assume `// legal statement, assigning String to an Object` should be _illegal_?

Reply:
Assigning a string to an object is legal. You're assigning a subclass of object to object.  
https://ideone.com/CwAzdi

Reply:
Also https://ideone.com/w82U99 ; not a `ClassCastException` since there's no generic types, but that's effectively the same error.